unpacker.go

```diff
@@ -226,16 +227,24 @@ func (u *unpacker) handlerWrapper(uctx context.Context, unpacks *int32) (func(im
 				if len(l) > 0 {
 					atomic.AddInt32(unpacks, 1)
 					eg.Go(func() error {
+						defer cancel()
```

Comment:
Won't this cause problems if there are multiple unpacks going on?

Reply:
How about this then? :)

Please note that if the unpacker returns without error, the context won't be cancelled, in theory the `updateCh` could still be full if there are 128 more unknown layers. :)